### PR TITLE
Upload `ruff` binaries to GitHub release

### LINKED ISSUE #2330: [REQUEST] Package standalone `ruff` binary & include in GitHub releases
**Context for the request**
I just came across `ruff` and it looks like a very promising pyflakes/isort replacement. We use Bazel to manage a large Python monorepo, and this seems like it could help shorten some critical dev loops and lead to a perf improvement in CI. 

However, given the way that ruff is packaged (ie. the main python entrypoint just shells out to the `ruff` binary under the hood), it doesn't seem like we'll be able to use `rules_python` + the `entry_point` macro from pip-parse to include ruff as a tool in our build (see https://github.com/bazelbuild/rules_python/issues/1000 for more discussion)

**Request**
OTOH, if ruff were available as a standalone binary (along w/ the source tgz/zip) as part of the GitHub release, we can pull this into our build w/ `http_archive`/set it up as an executable tool. This seems like it should be fairly straightforward to bake into the release process as a [release asset](https://docs.github.com/en/rest/releases/assets)

### PR DESCRIPTION
**Pull request by messense**
Closes #2330
Closes #2355

### CODE REVIEW

**Comment by messense** [comment]
Example run: https://github.com/messense/ruff/actions/runs/4185930263

Only the last `release` step isn't tested yet.

**Comment by messense** [comment]
BTW, added the `aarch64-pc-windows-msvc` target.

**Comment by charliermarsh** [comment]
Amazing!

File: .github/workflows/ruff.yaml [hunk 1]
```diff
@@ -63,26 +76,45 @@ jobs:
         with:
           name: wheels
           path: dist
+      - name: "Archive binary"
+        run: |
+            ARCHIVE_FILE=ruff-aarch64-apple-darwin.tar.gz
+            tar czvf $ARCHIVE_FILE -C target/aarch64-apple-darwin/release ruff
+            shasum -a 256 $ARCHIVE_FILE > $ARCHIVE_FILE.sha256
```
Comment (charliermarsh) [inline .github/workflows/ruff.yaml:83]: Do we need to strip the release binary, like in `ripgrep`?

https://github.com/BurntSushi/ripgrep/blob/44fb9fce2c1ee1a86c450702ea3ca2952bf6c5a7/.github/workflows/release.yml#L137

Comment (messense) [reply .github/workflows/ruff.yaml:83]: The binary isn't that big, I wouldn't bother strip it just yet, besides strip symbols makes panic backtrace hard to read.

I'd use `maturin build --strip` to also strip the binary in wheels if we want to strip the release binaries.

Comment (charliermarsh) [reply .github/workflows/ruff.yaml:83]: Okay cool, sounds good -- I wasn't sure.

Comment (messense) [reply .github/workflows/ruff.yaml:83]: https://github.com/charliermarsh/ruff/blob/d8e709648d2fe7c2bd7cf867fda1918bf6f092dc/pyproject.toml#L47

Actually it's already using `maturin build --strip`.

**Pull request merged**